### DISCUSSION TITLE
Can't seem to get rid of a space in my theme config

### DISCUSSION THREAD

**Discussion by georgefeakes** [JanDeDobbeleer/oh-my-posh#2870] [Troubleshoot] [answered]
I'm having some issues with spaces in my theme.

Could someone please let me know how to fix the space showing after my root segment before the trailing diamond?

![image](https://user-images.githubusercontent.com/34819831/193596860-b54cf5a2-7579-4e5b-9b59-671f36852465.png)

Also I'm trying to add in a space after the git branch icon before the name of the branch itself. Could someone please show me where this space needs to be added into the template.

![image](https://user-images.githubusercontent.com/34819831/193597097-f8f29efc-f623-4003-91d7-c142445668c0.png)

Here is my theme config:

```
{
  "$schema": "https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/schema.json",
  "blocks": [
    {
      "alignment": "left",
      "segments": [
        {
          "background": "#a313a8",
          "foreground": "#FFEB3B",
          "style": "diamond",
          "leading_diamond": "<transparent,#a313a8>\ue0b0</>",
          "trailing_diamond": "\ue0b0",
          "template": " \u26a1 ",
          "type": "root"
        },
        {
          "background": "#01579B",
          "foreground": "#ffffff",
          "leading_diamond": "<transparent,#01579B>\ue0b0</>",
          "properties": {
            "folder_icon": "\uf6d7",
            "folder_separator_icon": "<transparent> \ue0bd </>",
            "home_icon": "\uf7db",
            "style": "agnoster_short"
          },
          "style": "diamond",
          "template": " {{ .Path }} ",
          "trailing_diamond": "\ue0b0",
          "type": "path"
        },
        {
          "background": "#00C853",
          "background_templates": [
            "{{ if or (.Working.Changed) (.Staging.Changed) }}#FFEB3B{{ end }}",
            "{{ if and (gt .Ahead 0) (gt .Behind 0) }}#FFCC80{{ end }}",
            "{{ if gt .Ahead 0 }}#B388FF{{ end }}",
            "{{ if gt .Behind 0 }}#B388FF{{ end }}"
          ],
          "foreground": "#000000",
          "powerline_symbol": "\ue0b0",
          "properties": {
            "fetch_stash_count": true,
            "fetch_status": true
          },
          "style": "powerline",
          "template": " {{ .HEAD }}{{ if .Staging.Changed }}<#FF6F00> \uf046 {{ .Staging.String }}</>{{ end }}{{ if and (.Working.Changed) (.Staging.Changed) }} |{{ end }}{{ if .Working.Changed }} \uf044 {{ .Working.String }}{{ end }}{{ if gt .StashCount 0 }} \uf692 {{ .StashCount }}{{ end }} ",
          "type": "git"
        },
        {
          "background": "#49404f",
          "foreground": "#ffffff",
          "leading_diamond": "<transparent,#49404f>\ue0b0</>",
          "properties": {
            "style": "dallas",
            "threshold": 0
          },
          "style": "diamond",
          "template": " {{ .FormattedMs }}s ",
          "trailing_diamond": "\ue0b0",
          "type": "executiontime"
        },
        {
          "background": "#910000",
          "foreground": "#ffffff",
          "powerline_symbol": "\ue0b0",
          "style": "powerline",
          "template": "<transparent> \uf12a</> {{ .Meaning }} ",
          "type": "exit"
        }
      ],
      "type": "prompt"
    },
    {
      "alignment": "right",
      "segments": [
        {
          "background": "#29315A",
          "foreground": "#3EC669",
          "properties": {
            "time_format": "15:04:05"
          },
          "style": "diamond",
          "template": "<transparent>\ue0b0 </>{{ .CurrentDate | date .Format }} ",
          "type": "time"
        }
      ],
      "type": "prompt"
    },
    {
      "alignment": "left",
      "newline": true,
      "segments": [
        {
          "foreground": "#ffffff",
          "foreground_templates": [
            "{{ if gt .Code 0 }}#ff0000{{ end }}"
          ],
          "properties": {
            "always_enabled": true
          },
          "style": "plain",
          "template": "\u276f ",
          "type": "exit"
        }
      ],
      "type": "prompt"
    }
  ],
  "console_title_template": "{{if .Root}} \u26a1 {{end}}{{.Folder | replace \"~\" \"🏚\" }} @ {{.HostName}}",
  "version": 2
}
```

**Comment by lewis-yeung** [comment, marked as answer]
@georgefeakes

> Could someone please let me know how to fix the space showing after my root segment before the trailing diamond?

Are you using Windows Terminal? If so, you can update to its latest version which should have fixed the bug.

> Also I'm trying to add in a space after the git branch icon before the name of the branch itself. Could someone please show me where this space needs to be added into the template.

You can tweak `properties` of your `git` segment like this:

```json
"properties": {
  "branch_icon": "\ue0a0 ",
  "fetch_stash_count": true,
  "fetch_status": true
},
```

which means adding a space character after the original branch icon (`\ue0a0`).

**Reply by georgefeakes** [reply]
@lewis-yeung

> Are you using Windows Terminal? If so, you can update to its latest version which should have fixed the bug.

Yes, I'm using Windows Terminal. It seems to be up-to-date. No updates available in the Windows Store. Version is 1.15.2713.0

> You can tweak properties of your git segment like this...

This worked a treat! Thank you!

**Reply by JanDeDobbeleer** [reply]
@georgefeakes try replacing the `\u26a1` with another. It happens in Windows terminal that some icons are incorrectly measured causing this behaviour.

**Reply by georgefeakes** [reply]
@JanDeDobbeleer Ahhh interesting! Yes, it's caused by the icon rendering in Windows Terminal. Changing the icon to `\ueb53` ended up with the icon itself being clipped instead of the extra hidden space.

![image](https://user-images.githubusercontent.com/34819831/193649168-719deb31-eab0-4a69-a162-e7bf2db7a71f.png)

**Reply by JanDeDobbeleer** [reply]
@georgefeakes that's a bug for the Windows Terminal team in that case ;-)

**Reply by georgefeakes** [reply]
Got an icon that looks okay and renders correctly - thanks both! :)

![image](https://user-images.githubusercontent.com/34819831/193650393-f1fac4c5-bba5-4f86-9ec6-778020df6770.png)
